File: MediaScanner/src/main/java/com/app/scanner/MainActivity.java
package com.app.scanner;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;

import android.os.Handler;
import android.os.storage.*;

import android.util.Log;
import androidx.annotation.Nullable;

import com.app.scanner.db.*;

import com.app.scanner.util.LogUtils;

import java.util.ArrayList;
import java.util.List;

/**********************************************
 * Filename： MainActivity
 * Author:   wangyi@zlingsmart.com.cn
 * Description：
 * Date：
 * Version:
 * History:
 *------------------------------------------------------
 * Version  date      author   description
 * V0.0.1           1) …
 ***********************************************/
public class MainActivity extends Activity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main_layout);
//        DeviceManager.getInstance(this);
//
//        StorageManager mStorageManager = (StorageManager) CarApp.contextApp.getSystemService(Context.STORAGE_SERVICE);
//
//        mStorageManager.registerListener(new StorageEventListener(){
//            @Override
//            public void onUsbMassStorageConnectionChanged(boolean connected) {
//                super.onUsbMassStorageConnectionChanged(connected);
//                LogUtils.err("onUsbMassStorageConnectionChanged");
//            }
//
//            @Override
//            public void onStorageStateChanged(String path, String oldState, String newState) {
//                super.onStorageStateChanged(path, oldState, newState);
//                LogUtils.err("onStorageStateChanged");
//            }
//
//            @Override
//            public void onVolumeStateChanged(VolumeInfo vol, int oldState, int newState) {
//                super.onVolumeStateChanged(vol, oldState, newState);
//                LogUtils.err("onVolumeStateChanged");
//            }
//
//            @Override
//            public void onVolumeRecordChanged(VolumeRecord rec) {
//                super.onVolumeRecordChanged(rec);
//                LogUtils.err("onVolumeRecordChanged");
//            }
//
//            @Override
//            public void onVolumeForgotten(String fsUuid) {
//                super.onVolumeForgotten(fsUuid);
//
//                LogUtils.err("onVolumeForgotten");
//            }
//
//            @Override
//            public void onDiskScanned(DiskInfo disk, int volumeCount) {
//                super.onDiskScanned(disk, volumeCount);
//
//                LogUtils.err("onDiskScanned");
//            }
//
//            @Override
//            public void onDiskDestroyed(DiskInfo disk) {
//                super.onDiskDestroyed(disk);
//                LogUtils.err("onDiskDestroyed");
//            }
//        });


        final DbOperationHelper helper = new DbOperationHelper(AudioVo.class,
                DaoManager.getInstance().getDaoSession().getAudioVoDao());
        List<AudioVo> tempList = new ArrayList<>();
        for (int i = 0; i < 20000; i++) {
            AudioVo albumVo = new AudioVo();
            albumVo.setName("1111");
            albumVo.setSymbolName("1");
            albumVo.setDuration("1");
            albumVo.setFavFlag("1");
            albumVo.setSize("1");
            albumVo.setFolderId(1l);
            albumVo.setAlbumId(1l);
            albumVo.setPath("111111");
            albumVo.setYear("1");
            tempList.add(albumVo);
        }
        long temp1 = System.currentTimeMillis();
        boolean flag = helper.insertMultiple(tempList);
        long temp2 = System.currentTimeMillis() - temp1;
        List<AudioVo> list = helper.queryAll();
        Log.d("MainActivity666", list.size() + " ;" + flag + "  ;" + temp2);

    }
}

File: MediaScanner/src/main/java/com/app/scanner/MainActivity.kt
package com.app.scanner

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.widget.TextView
import android.widget.Toast
import com.app.scanner.db.DaoManager
import com.app.scanner.db.DbOperationHelper
import com.app.scanner.util.LogUtils
import com.app.scanner.vo.AudioVo


/**********************************************
 * Filename： MainActivity
 * Author:   wangyi@zlingsmart.com.cn
 * Description：
 * Date：
 * Version:
 * History:
 * ------------------------------------------------------
 * Version  date      author   description
 * V0.0.1           1) …
 */
class MainActivity : Activity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main_layout)



        val helper: DbOperationHelper<AudioVo> = DbOperationHelper<AudioVo>(
            AudioVo::class.java,
            DaoManager.getInstance().daoSession.audioVoDao
        )

        findViewById<TextView>(R.id.tv_usb1_in).setOnClickListener {
            LogUtils.debug("click tv_usb1_in")
            var intent = Intent()
            intent.action = ScannerService.MOCK_IN
            sendBroadcast(intent)
        }

        findViewById<TextView>(R.id.tv_usb1_out).setOnClickListener {
            var intent = Intent()
            intent.action = ScannerService.MOCK_OUT
            sendBroadcast(intent)
        }

        findViewById<TextView>(R.id.tv_insert).setOnClickListener {
            val albumVo = AudioVo()
            albumVo.name = "1111"
            albumVo.symbolName = "1"
            albumVo.duration = "1"
            albumVo.favFlag = "1"
            albumVo.size = "1"
            albumVo.folderId = 1L
            albumVo.albumId = 1L
            albumVo.path = "111111"
            albumVo.year = "1"
            albumVo.genreId = 1L
            helper.insert(albumVo)
            Toast.makeText(this, "已写入数据", Toast.LENGTH_SHORT).show()
        }

        findViewById<TextView>(R.id.tv_query).setOnClickListener {
            val list: List<AudioVo> = helper.queryAll()
            Toast.makeText(this, "当前数据量：" + list.size, Toast.LENGTH_SHORT).show()
        }

        findViewById<TextView>(R.id.tv_get).setOnClickListener {
            val list: List<AudioVo> = helper.queryAll()
            if (list.size > 0) {
                Toast.makeText(this, "歌名：" + list.get(0).name, Toast.LENGTH_SHORT).show()
            } else {
                Toast.makeText(this, "没有数据", Toast.LENGTH_SHORT).show()
            }
        }

        findViewById<TextView>(R.id.tv_update).setOnClickListener {
            val list: List<AudioVo> = helper.queryAll()
            if (list.size > 0) {
                var teml = list.get(0)
                teml.name = "七里香"
                helper.update(teml)
                Toast.makeText(this, "数据已经更新", Toast.LENGTH_SHORT).show()
            } else {
                Toast.makeText(this, "没有数据", Toast.LENGTH_SHORT).show()
            }
        }

        findViewById<TextView>(R.id.tv_delete).setOnClickListener {
            val list: List<AudioVo> = helper.queryAll()
            if (list.size > 0) {
                var teml = list.get(0)
                helper.delete(teml)
                Toast.makeText(this, "数据已经删除", Toast.LENGTH_SHORT).show()
            } else {
                Toast.makeText(this, "没有数据", Toast.LENGTH_SHORT).show()
            }
        }

        findViewById<TextView>(R.id.tv_skip_jni).setOnClickListener {
            var intent = Intent()
            intent.setClass(this, TestJniMainAct::class.java)
            startActivity(intent)
        }

    }

}